site/en/blog/private-network-access-preflight/index.md

```diff
@@ -0,0 +1,334 @@
+
+```text
+HTTP/1.1 PUT /give-me-root
+Origin: https://foo.example
+```
+
+To which the server can respond per usual CORS rules:
+
+```text
+HTTP/1.1 200 OK
+Access-Control-Allow-Origin: https://foo.example
+```
+
+## How to know if your website is affected
+
+Starting in Chrome 98, if a private network request is detected, a preflight
+request will be sent ahead of it. If this preflight request fails, the final
+request will still be sent, but a warning will be surfaced in the DevTools
+issues panel.
+
+{% Img
+   src="image/VbsHyyQopiec0718rMq2kTE1hke2/AgZzPf3NkMWQ0Cm6Puu0.png",
+   alt="A failed preflight request warning in the Devtools Issues panel",
+   class="screenshot",
+   width="800", height="556"
+%}
+
+Affected preflight requests can also be viewed and diagnosed in the network panel:
+
+{% Img
+   src="image/VbsHyyQopiec0718rMq2kTE1hke2/aysOX5wKA1kme8HyV3t0.png",
+   alt="A failed preflight request in the DevTools Network panel",
+   class="screenshot",
+   width="800", height="265"
+%}
+
+To see what happens if preflight success was enforced, you can
+[pass the following command-line argument](https://www.chromium.org/developers/how-tos/run-chromium-with-flags)
+starting in Chrome 98:
+
+```text
+–enable-features=PrivateNetworkAccessRespectPreflightResutls
+```
+
+Any failed preflight request will result in a failed fetch. This can allow you
+to test whether your website would work after the
+[second phase of our rollout plan](#rollout-plan). Errors can be diagnosed in
+the same way as warnings using the DevTools panels mentioned above.
+
+## What to do if your website is affected
+
+While this change is not expected to break any website, we strongly encourage
+you to get ahead of the [second phase of our rollout](#rollout-plan) and start
+fixing affected request paths as soon as possible.
+
+There are two solutions available to you:
+
+1. Handle preflight requests on the server side
+1. Disable PNA checks with enterprise policies
+
+### Handle preflight requests server-side {: #server-side-requests }
+
+Update the target server of any affected fetches to handle PNA preflight
+requests. First, implement support for standard CORS preflight requests on
+affected routes. Then add support for the [two new response headers](#new-in-pna).
+
+When your server receives a preflight request (an `OPTIONS` request with CORS
+headers), it should check for the presence of an
+`Access-Control-Request-Private-Network: true` header. If this header is
+present on the request, the server should examine the `Origin` header and the
+request path, and any other relevant information (such as
+`Access-Control-Request-Headers`) to make sure the request is safe to allow.
+Typically, you will want to allow access to a single origin under your control.
+Beware of insecure (non-https) origins, as they are unauthenticated: an on-path
+attacker could masquerade as any such origin!
+
+Once your server has decided to allow the request, it should respond 200 OK
```

Comment:
For consistency with the example above, we should recommend 204 here. It turns out that some smart TVs etc. do not like that, so 200 OK also works. There seems to be a small debate about which one is correct, but both can work.

```suggestion
Once your server has decided to allow the request, it should respond 204 No Content (or 200 OK)
```